src/v/storage/tests/index_state_test.cc

```diff
@@ -149,9 +149,14 @@ BOOST_AUTO_TEST_CASE(serde_deprecated_crc) {
 
     BOOST_REQUIRE_EXCEPTION(
       serde::from_iobuf<storage::index_state>(bad_buf.copy()),
-      serde::serde_exception,
-      [](const serde::serde_exception& e) {
-          return std::string_view(e.what()).find("Invalid checksum for index")
-                 != std::string_view::npos;
+      std::exception,
+      [](const std::exception& e) {
+          auto is_crc = std::string_view(e.what()).find(
+                          "Invalid checksum for index")
+                        != std::string_view::npos;
+          auto is_out_of_bounds = std::string_view(e.what()).find(
+                                    "Invalid consume_to")
+                                  != std::string_view::npos;
```

Comment:
nit, how about defining a `msg`? less repeatings this way:
```suggestion
          auto msg = std::string_view(e.what());
          auto is_crc = msg.find(
                          "Invalid checksum for index")
                        != msg.npos;
          auto is_out_of_bounds = msg.find(
                                    "Invalid consume_to")
                                  != msg.npos;
```

Reply:
done, thank you